src/openrct2-ui/windows/GameBottomToolbar.cpp

```diff
@@ -333,11 +331,9 @@ static void window_game_bottom_toolbar_invalidate(rct_window* w)
         w->disabled_widgets &= ~(1 << WIDX_NEWS_LOCATE);
 
         // Find out if the news item is no longer valid
-        int32_t y, z;
-        int32_t subject = newsItem->Assoc;
-        news_item_get_subject_location(newsItem->Type, subject, &x, &y, &z);
+        auto subjectLoc = news_item_get_subject_location(newsItem->Type, newsItem->Assoc);
 
-        if (x == LOCATION_NULL)
+        if (!subjectLoc)
```

Comment:
I prefer an explicit comparison (with nullopt), in the same way we explicitly compare to nullptr.